File: examples/cpp/buttons.cpp
/**
 * Example using LED with pigpio
 *
 * sudo apt-get install pigpio
 * make
 * sudo ./led
 */
#include <iostream>
#include <vector>
#include <pigpio.h>

int main()
{
	if(gpioInitialise() == 0)
		return -1;

	std::vector<unsigned int> leds = { 21, 20, 16, 12, 7, 8 };
	unsigned int led_state = 0;
	for(auto const i: leds)
		gpioSetMode(i,PI_INPUT);

	std::vector<unsigned int> buttons = {5, 6, 19, 13, 4, 17};
	std::vector<unsigned int> button_state = {0, 0, 0, 0, 0, 0};
	for(auto const i: buttons)
		gpioSetMode(i,PI_INPUT);

	gpioSetMode(14,PI_INPUT);
	gpioSetMode(15,PI_INPUT);
	gpioSetMode(18,PI_INPUT);

	while(true)
	{
		for(size_t i = 0; i < buttons.size(); ++i)
		{
			if(gpioRead(buttons[i])) {
				if(button_state[i] == 0) {
					button_state[i] = 1;
					std::cout << "Switch "  << i+1 << " pressed" << std::endl;

					gpioSetMode(leds[led_state],PI_INPUT);
					led_state = i;
					gpioSetMode(leds[led_state],PI_OUTPUT);
				}
			} else {
				if(button_state[i] == 1) {
					button_state[i] = 0;
					std::cout << "Switch "  << i+1 << " released" << std::endl;
				}
			}
		}

		gpioSleep(PI_TIME_RELATIVE, 0, 10 * 1000);
	}

	gpioTerminate();
	return 0;
}

File: examples/buttons.py
import RPi.GPIO as IO
import time

IO.setmode(IO.BCM)

buttons = [5,6,19,13,4,17]
button_state = [0] * 8

leds = [21, 20, 16, 12, 7, 8]
led_state = 0

for led in leds:
	IO.setup(led, IO.IN)
IO.setup(leds[led_state%6], IO.OUT)

for i in range(0, 6):
	print('Setup switch ', i+1, ' pin ' , buttons[i])
	IO.setup(buttons[i], IO.IN, pull_up_down=IO.PUD_DOWN)

while 1: 
	for i in range(0, 6):
		if IO.input(buttons[i]): 
			if button_state[i] == 0:
				button_state[i] = 1
				print('Switch ', i+1, 'pressed')

				IO.setup(leds[led_state%6], IO.IN)
				led_state = i
				IO.setup(leds[led_state%6], IO.OUT)

		else:
			if button_state[i] == 1:
				new_led_state = 18 + random.randint(0, 6)
				button_state[i] = 0
				print('Switch ', i+1, 'released')

				
	time.sleep(0.01)

